tests/error/nightly/mod.rs

```diff
@@ -86,6 +86,8 @@ impl Default for BacktraceErr {
 
 impl Error for BacktraceErr {
     fn provide<'a>(&'a self, demand: &mut std::any::Demand<'a>) {
-        demand.provide_ref(&self.backtrace).provide_value(|| 42);
+        demand
+            .provide_ref::<Backtrace>(&self.backtrace)
+            .provide_value::<i32>(42);
```

Comment:
@tyranron downside of using nightly is that [API can change at any time][1] 🙃 

[1]: https://github.com/rust-lang/rust/pull/99583/files#diff-0752889661748b8a15a597d7156127b6fb90fdeda0627be50e07f3f785bd0f4dR799

Reply:
@ilslv let's see for a while how it will play.